Dependencies/xr/Source/OpenXR/XR.cpp

```diff
@@ -985,19 +985,20 @@ namespace xr
             swapchainCreateInfo.createFlags = createFlags;
             swapchainCreateInfo.usageFlags = usageFlags;
 
-            XrSecondaryViewConfigurationSwapchainCreateInfoMSFT secondaryViewConfigCreateInfo
-            {
-                XR_TYPE_SECONDARY_VIEW_CONFIGURATION_SWAPCHAIN_CREATE_INFO_MSFT
-            };
-            if (HmdImpl.Context.Extensions()->SecondaryViewConfigurationSupported)
+            if (HmdImpl.Context.Extensions()->SecondaryViewConfigurationSupported && 
+                viewConfigType == XR_VIEW_CONFIGURATION_TYPE_SECONDARY_MONO_FIRST_PERSON_OBSERVER_MSFT)
             {
+                XrSecondaryViewConfigurationSwapchainCreateInfoMSFT secondaryViewConfigCreateInfo
```

Comment:
This variable need to be moved out of the if block, to keep it at same lifetime as the xrCreateSwapchain function below.  Otherwise, the memory in the stack might be overwritten when the function is called below.